app/models/manageiq/providers/amazon/instance_types.rb

```diff
@@ -14,11 +14,11 @@ module ManageIQ::Providers::Amazon::InstanceTypes
       :ebs_only                => true,
       :instance_store_size     => 0,
       :instance_store_volumes  => 0,
-      :architecture            => [:x86_64],
+      :architecture            => [:i386, :x86_64],
       :virtualization_type     => [:hvm],
       :network_performance     => :low,
       :physical_processor      => "Intel Xeon family",
-      :processor_clock_speed   => 3.3,
+      :processor_clock_speed   => "up to 3.3",
```

Comment:
is this a problem, that this is now a string?
is the db a string?
will it impact UI sorting?

Reply:
The original code was 

`o[:processor_clock_speed] = row[:clock_speed_ghz].scan(/[\d\.]/).join`

And that won't work because some of the new values of `row[:clock_speed_ghz]` are like `'2.3 to 2.7'`. The `scan` will return '2.32.7' which doesn't make sense and also breaks the numeric field.

This is the part need your help. Do you know how we use this? Or who would know that I can ask?

Reply:
Actually this won't matter, the db doesn't care about this, it only cares about 
```
new_result = {
      :type                     => ManageIQ::Providers::Amazon::CloudManager::Flavor.name,
      :ems_ref                  => uid,
      :name                     => name,
      :description              => flavor[:description],
      :enabled                  => !flavor[:disabled],
      :cpus                     => cpus,
      :cpu_cores                => 1,
      :memory                   => flavor[:memory],
      :supports_32_bit          => flavor[:architecture].include?(:i386),
      :supports_64_bit          => flavor[:architecture].include?(:x86_64),
      :supports_hvm             => flavor[:virtualization_type].include?(:hvm),
      :supports_paravirtual     => flavor[:virtualization_type].include?(:paravirtual),
      :block_storage_based_only => flavor[:ebs_only],
      :cloud_subnet_required    => flavor[:vpc_only],
      :ephemeral_disk_size      => flavor[:instance_store_size],
      :ephemeral_disk_count     => flavor[:instance_store_volumes]
    }
```
as in [parsing code here](https://github.com/jameswnl/manageiq-providers-amazon/blob/32bc2aefe1f39b86030000031f615dde650a97ae/app/models/manageiq/providers/amazon/cloud_manager/refresh_parser.rb#L113-L113)